technico-v16/app/components/AuthContext.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import { createContext, ReactNode, useContext, useState, useEffect } from 'react';
+import { useRouter } from 'next/navigation';
+
+interface AuthStatus {
+  userId: string;
+  isAuthenticated: boolean;
+  isAdmin: boolean;
+}
+
+interface AuthContextType extends AuthStatus {
+  fireReload: () => void;
+}
+
+export const AuthContext = createContext<AuthContextType | null>(null);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [auth, setAuth] = useState<AuthStatus>({ isAuthenticated: false, isAdmin: false, userId: "" });
+  const [reload, setReload] = useState(false);
+  const router = useRouter();
+
+  const fireReload = () => {
+    setReload((prevReload) => !prevReload);
+  };
+
```

Comment:
-   Good use of useEffect for auth state management
-   Good handling of different token types and routing logic
-   Consider extracting token validation to a separate utility function